source/s3_auto_ranged_put.c

```diff
@@ -990,9 +1076,18 @@ struct aws_future *s_s3_prepare_upload_part(struct aws_s3_request *request) {
          * Next async step: read through the body stream until we've
          * skipped over parts that were already uploaded (in case we're resuming
          * from an upload that had been paused) */
-        part_prep->skipping_future = s_skip_parts_from_stream(
-            meta_request, auto_ranged_put->prepare_data.num_parts_read_from_stream, request->part_number - 1);
-        aws_future_register_callback(part_prep->skipping_future, s_s3_prepare_upload_part_on_skipping_done, part_prep);
+        if (auto_ranged_put->has_content_length) {
+            part_prep->skipping_future = s_skip_parts_from_stream(
```

Comment:
Let's just call `s_skip_parts_from_stream()` even if we don't have content length

It doesn't do any work unless we're resuming from a pause/resume, and we don't even let a meta-request get created with pause/resume/no Content-Length

this function was already one we always called, expecting that most of the time it would do no work

just calling it here seems simpler than the whole fake-future thing we do in the else statement